### PR TITLE
[FX-4614] Migrate Link

### PR DESCRIPTION
**Pull request by mkrl**
[FX-4614](https://toptal-core.atlassian.net/browse/FX-4614)

### Description

Migrates Link away from MUI to pure `Typography` + Tailwind implementation

### How to test

- [Temploy](https://picasso.toptal.net/fx-4614-migrate-link)
- No visual changes should be present
- The code looks good

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4614]: https://toptal-core.atlassian.net/browse/FX-4614?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .changeset/curly-ants-wave.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@toptal/picasso-link': major
```
Comment (TomasSlama) [inline .changeset/curly-ants-wave.md:2]:
```suggestion
'@toptal/picasso-link': major
'@toptal/picasso': patch
```

File: packages/base/Link/package.json [hunk 2]
```diff
@@ -23,7 +23,8 @@
   "homepage": "https://github.com/toptal/picasso/tree/master/packages/picasso#readme",
   "dependencies": {
     "@toptal/picasso-utils": "1.0.3",
-    "classnames": "^2.5.1"
+    "@toptal/picasso-typography": "2.0.1",
+    "tailwind-merge": "^2.2.2"
```
Comment (TomasSlama) [inline packages/base/Link/package.json:27]: we are missing `@toptal/picasso-tailwind` peer dependency and also we can probably remove the `@material-ui/core`

**Review by TomasSlama — COMMENTED** [review]
It seems there are some related changes in screenshots

File: packages/base/Link/src/Link/Link.tsx [hunk 3]
```diff
@@ -80,32 +100,38 @@ export const Link: OverridableComponent<Props> = forwardRef<
...
+      className={twMerge(
+        'focus:outline-none hover:underline',
+        COLOR_DISABLED_MAP[color][variant][disabled ? 'disabled' : 'normal'],
+        disabled ? 'cursor-not-allowed' : '',
+        noUnderline ? '!no-underline' : '',
+        visited
+          ? color === 'blue'
+            ? 'visited text-purple-500'
+            : 'visited text-gray-500'
+          : '',
```
Comment (OleksandrNechai) [inline packages/base/Link/src/Link/Link.tsx:126]: Too many ternaries to my taste. Maybe another MAP would make it simpler?

**Review by OleksandrNechai — APPROVED** [review]
Code looks good to me. Did not test, but I guess we should rely on auto tests that we have and checks in portals.

**Pull request merged**